server/controllers/auth/gymLogin.ts

```diff
@@ -0,0 +1,57 @@
+import { NextFunction, Response, Request } from 'express';
+
+import { gymLoginSchema, CustomError, generateToken, comparePassword } from '../../utils';
+import { Gym } from '../../database/models';
+
+export default async function gymLogin(req: Request, res: Response, next: NextFunction) {
+  try {
+    // Validate the request body against the schema
+    const {
+      email,
+      password: gymPassword,
+    } = await gymLoginSchema.validateAsync(req.body);
+
+    // Check if the gym already exists
+    const isExist: any = await Gym.findOne({
+      where: { email },
+    });
+
+    // if is not exist throw an error
+    if (!isExist) {
+      throw new CustomError('عذرا لا يوجد حساب نادي بهذا البريد الالكتروني! حاول مرة أخرى', 401);
+    }
+
+    const { id, gymName, password: hashedPassword } = isExist;
+
+    const checkedPassword: boolean = await comparePassword(gymPassword, hashedPassword);
```

Comment:
we are removing Function comparePassword So, 

it would be best if you used **compare ** form bycript.

`import { compare } from 'bcryptjs';`

```suggestion
    const checkedPassword: boolean = await compare (...);
```